pipelines/batch/src/main/java/com/google/fhir/analytics/FhirSearchUtil.java

```diff
@@ -68,8 +68,8 @@ public Bundle searchByUrl(String searchUrl, int count, SummaryEnum summaryMode)
       return result;
     } catch (Exception e) {
       log.error("Failed to search for url: " + searchUrl + " ;  " + "Exception: " + e);
+      throw e;
```

Comment:
I am assuming this should fail the pipeline altogether, retrying the http request after a delay is a good to have feature. Regardless, the missing records issue will not occur after this.

Reply:
Thanks for looking into this. on point 2) I agree that 200 second is a fairly large time, On the other hand I was also thinking since the FHIR server did not respond with in a given time (60 secs) due to some overload at its end, may be the client can still do a retry after a delay (can be exponential back off with a fixed number of retries) hoping that the server can now respond (we might have to handle the SocketTimeoutException here as it is a non retriable error currently). I also assume in reality there would be some kind of circuit breaker on the FHIR server which will temporarily open the circuit in this case due to lot of timeouts and redirect the clients quickly (no waiting) to a busy error (http code 429) so that the the clients can retry after a while.

Reply:
Even I am not aware on point 3) why the failed worker is not retried. May be we should analyze and document on this behaviour.